File: Aplicación Escritorio/ModeloTDO/RutaDTO.cs
using Modelo;
using System.Runtime.InteropServices;
using System.Text.Json.Serialization;

namespace ModeloDTO
{
    public class RutaDTO
    {
        public long idRuta { get; set; }
        public long idUsuario { get; set; }
        public string nombre { get; set; }
        public string nombreInicioruta { get; set; }
        public string nombreFinalruta { get; set; }

        public double latitudInicial { get; set; }
        public double latitudFinal { get; set; }
        public double longitudInicial { get; set; }
        public double longitudFinal { get; set; }
        public double distancia { get; set; }

        public TimeSpan duracion { get; set; }
        public int desnivelPositivo { get; set; }
        public int desnivelNegativo { get; set; }

        public int desnivelAcumulado { get; set; }
        public double altitudMax { get; set; }
        public double altitudMin { get; set; }
        public byte nivelEsfuerzo { get; set; }
        public byte nivelRiesgo { get; set; }
        public byte tipoTerreno { get; set; }
        public byte indicaciones{ get; set; }
        public string temporadas { get; set; }
        public string recomendacionesEquipo { get; set; }
        public double? mediaEstrellas { get; set; }
        public string zonaGeografica { get; set; }
        public bool estadoRuta { get; set; }
        public bool accesible { get; set; }
        public bool familiar { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public CLASIFICACION clasificacion { get; set; }
    }
    }

File: Springboot_API/RetaCantabria/src/main/java/org/example/DTO/RutaDTO.java
package org.example.DTO;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import org.example.Entidades.CLASIFICACION;
import org.example.Entidades.Ruta;

import java.time.LocalTime;

public record RutaDTO(
        Long idRuta,
        Long idUsuario,
        String nombre,
        String nombreInicioruta,
        String nombreFinalruta,

        double latitudInicial,
        double latitudFinal,
        double longitudInicial,
        double longitudFinal,
        Double distancia,
        @JsonFormat(pattern = "HH:mm:ss")
        LocalTime duracion,
        Integer desnivelPositivo,
        Integer desnivelNegativo,

        Integer desnivelAcumulado,
        Double altitudMax,
        Double altitudMin,
        Byte nivelEsfuerzo,
        Byte nivelRiesgo,
        Byte tipoTerreno,
        Byte indicaciones,
        String temporadas,
        String recomendacionesEquipo,
        Double mediaEstrellas,
        String zonaGeografica,
        Boolean estadoRuta,
        Boolean accesible,
        Boolean familiar,
        @Enumerated(EnumType.STRING)
        CLASIFICACION clasificacion
) {
    public RutaDTO(Ruta r) {
        this(
                r.getIdRuta(),
                r.getUsuarioIdusuario().getIdUsuario(),
                r.getNombre(),
                r.getNombreInicioruta(),
                r.getNombreFinalruta(),
                r.getLatitudInicial(),
                r.getLatitudFinal(),
                r.getLongitudInicial(),
                r.getLongitudFinal(),
                r.getDistancia(),
                r.getDuracion(),
                r.getDesnivelPositivo(),
                r.getDesnivelNegativo(),
                r.getDesnivelAcumulado(),
                r.getAltitudMax(),
                r.getAltitudMin(),
                r.getNivelEsfuerzo(),
                r.getNivelRiesgo(),
                r.getTipoTerreno(),
                r.getIndicaciones(),
                r.getTemporadas(),
                r.getRecomendacionesEquipo(),
                r.getMediaEstrellas(),
                r.getZonaGeografica(),
                r.isEstadoRuta(),
                r.isRutaFamiliar(),
                r.isRutaFamiliar(),
                r.getClasificacion()
        );
    }
}
